sdk/core/src/http_client.rs

```diff
@@ -62,69 +74,126 @@ pub trait HttpClient: Send + Sync + std::fmt::Debug {
+//        let mut hyper_request = hyper::Request::builder()
+//            .uri(request.uri())
+//            .method(request.method());
+//
+//        for header in request.headers() {
+//            hyper_request = hyper_request.header(header.0, header.1);
+//        }
+//
+//        let hyper_request = hyper_request.body(hyper::Body::from(request.into_body()))?;
+//
+//        let hyper_response = self.request(hyper_request).await?;
+//
+//        let mut response = Response::builder()
+//            .status(hyper_response.status())
+//            .version(hyper_response.version());
+//
+//        for (key, value) in hyper_response.headers() {
+//            response = response.header(key, value);
+//        }
+//
+//        let response = response.body(hyper::body::to_bytes(hyper_response.into_body()).await?)?;
+//
+//        Ok(response)
+//    }
+//}
+
+#[cfg(feature = "enable_reqwest")]
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
-impl HttpClient for hyper::Client<HttpsConnector<hyper::client::HttpConnector>> {
+impl HttpClient for reqwest::Client {
     async fn execute_request(
         &self,
         request: Request<Bytes>,
     ) -> Result<Response<Bytes>, Box<dyn std::error::Error + Sync + Send>> {
-        let mut hyper_request = hyper::Request::builder()
-            .uri(request.uri())
-            .method(request.method());
-
+        let mut reqwest_request = self.request(
+            request.method().clone(),
+            url::Url::parse(&request.uri().to_string()).unwrap(),
+        );
         for header in request.headers() {
-            hyper_request = hyper_request.header(header.0, header.1);
+            reqwest_request = reqwest_request.header(header.0, header.1);
         }
 
-        let hyper_request = hyper_request.body(hyper::Body::from(request.into_body()))?;
+        let reqwest_request = reqwest_request.body(request.into_body()).build()?;
 
-        let hyper_response = self.request(hyper_request).await?;
+        let reqwest_response = self.execute(reqwest_request).await?;
 
         let mut response = Response::builder()
-            .status(hyper_response.status())
-            .version(hyper_response.version());
+            .status(reqwest_response.status())
+            .version(reqwest_response.version());
 
-        for (key, value) in hyper_response.headers() {
+        for (key, value) in reqwest_response.headers() {
             response = response.header(key, value);
         }
 
-        let response = response.body(hyper::body::to_bytes(hyper_response.into_body()).await?)?;
+        let response = response.body(reqwest_response.bytes().await?)?;
 
         Ok(response)
     }
-}
 
-#[cfg(feature = "enable_reqwest")]
-#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
-#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
-impl HttpClient for reqwest::Client {
-    async fn execute_request(
+    async fn execute_request2(
```

Comment:
Can we use a more explanitory name here, like `execute_request_raw` or keep this as `execute_request` and make the other `execute_bytes_request` or `execute_request_with_bytes`? Names like this don't really help developers at a glance know what they do, and we use a similar approach in other languages were overloads aren't supported (JS, Python).

Reply:
There's a comment above explaining this is very temporary. We will rename this to `execute_request` as soon as all requests are moved to using the new pipeline infrastructure. That should hopefully be quickly after merging this PR.